zaza/openstack/charm_tests/tempest/setup.py

```diff
@@ -0,0 +1,378 @@
+    # https://docs.openstack.org/networking-ovn/2.0.1/features.html
+    # This enables test_list_show_extensions to run successfully.
+    if current_release >= focal_ussuri:
+        extensions = ('address-scope,agent,allowed-address-pairs,'
+                      'auto-allocated-topology,availability_zone,'
+                      'binding,default-subnetpools,external-net,'
+                      'extra_dhcp_opt,multi-provider,net-mtu,'
+                      'network_availability_zone,network-ip-availability,'
+                      'port-security,provider,quotas,rbac-address-scope,'
+                      'rbac-policies,standard-attr-revisions,security-group,'
+                      'standard-attr-description,subnet_allocation,'
+                      'standard-attr-tag,standard-attr-timestamp,trunk,'
+                      'quota_details,router,extraroute,ext-gw-mode,'
+                      'fip-port-details,pagination,sorting,project-id,'
+                      'dns-integration,qos')
+        ctxt['neutron_api_extensions'] = extensions
+    else:
+        ctxt['neutron_api_extensions'] = 'all'
+
+
+def add_glance_config(ctxt, keystone_session):
+    """Add glance config to context.
+
+    :param ctxt: Context dictionary
+    :type ctxt: dict
+    :returns keystone_session: keystoneauth1.session.Session object
+    :type: keystoneauth1.session.Session
+    :returns: None
+    :rtype: None
+    """
+    glance_client = openstack_utils.get_glance_session_client(
+        keystone_session)
+    image = openstack_utils.get_images_by_name(
+        glance_client, glance_setup.CIRROS_IMAGE_NAME)
+    image_alt = openstack_utils.get_images_by_name(
+        glance_client, TEMPEST_CIRROS_ALT_IMAGE_NAME)
+    if image:
+        ctxt['image_id'] = image[0].id
+    if image_alt:
+        ctxt['image_alt_id'] = image_alt[0].id
+
+
+def add_cinder_config(ctxt, keystone_session):
+    """Add cinder config to context.
+
+    :param ctxt: Context dictionary
+    :type ctxt: dict
+    :returns keystone_session: keystoneauth1.session.Session object
+    :type: keystoneauth1.session.Session
+    :returns: None
+    :rtype: None
+    """
+    volume_types = ['volumev2', 'volumev3']
+    keystone_client = openstack_utils.get_keystone_session_client(
+        keystone_session)
+    for volume_type in volume_types:
+        service = keystone_client.services.list(type=volume_type)
+        if service:
+            ctxt['catalog_type'] = volume_type
+            break
+
+
+def add_keystone_config(ctxt, keystone_session):
+    """Add keystone config to context.
+
+    :param ctxt: Context dictionary
+    :type ctxt: dict
+    :returns keystone_session: keystoneauth1.session.Session object
+    :type: keystoneauth1.session.Session
+    :returns: None
+    :rtype: None
+    """
+    keystone_client = openstack_utils.get_keystone_session_client(
+        keystone_session)
+    for domain in keystone_client.domains.list():
+        if domain.name == 'admin_domain':
```

Comment:
This is nitpicky but you can use something like

keystone_client.domains.find(name="admin_domain") rather than the for loop.